language/testing/language.go

```diff
@@ -85,6 +86,10 @@ func (tc *TestCaseMistakes) Validate(t *testing.T) {
 		actualMistakes, actualErr := tc.Language.Mistakes(logger, repositoryPath)
 		require.NoError(t, actualErr)
 
-		assert.Equal(t, tc.ExpectedMistakes, actualMistakes)
+		if tc.ExpectedMistakesContains != nil {
+			tc.ExpectedMistakesContains(actualMistakes)
+		} else {
+			assert.Equal(t, tc.ExpectedMistakes, actualMistakes)
+		}
```

Comment:
Please always hand through the testing object! This can get very confusing if assertions fail. See here:

```go
func TestSomething(tOuter *testing.T) {
    ...
    validate(tOuter, &testCase{
        Name: "FOO"
        SomeFunctionWithLogic: func(tInner *testing.T){
            assert.True(tOuter, false, "messageOuter")
            // If this fails, the error message will show up with just "TestSomething" cause it is the outer T instance.
            
            assert.True(tInner, false, "messageInner")
            // If this fails, the error message will show up with "TestSomething/FOO" and we know exactly where it comes from.
        }
    })
}
```